packages/xrpl/src/models/utils/flags.ts

```diff
@@ -97,22 +113,24 @@ function convertFlagsToNumber(flags: GlobalFlags, flagEnum: any): number {
  * @returns A map with all flags as booleans.
  */
 export function parseTransactionFlags(tx: Transaction): object {
-  setTransactionFlagsToNumber(tx)
-  if (typeof tx.Flags !== 'number' || !tx.Flags || tx.Flags === 0) {
+  const flags = convertTxFlagsToNumber(tx)
+  if (flags === 0) {
```

Comment:
there is value in retaining the old checks. `convertTxFlagsToNumber` method does not retain the exact order of these checks in its current implementation.

Reply:
All of the same checks still exist in the current logic, just in a different representation and in the helper method. I have extracted the null check to be earlier now so that it is clearer that it is the same